src/rez/package_cache.py

```diff
@@ -460,12 +471,14 @@ def add_variants_async(self, variants):
                 else:
                     out_target = devnull
 
-                subprocess.Popen(
-                    [exe, "--daemon", self.path],
```

Comment:
Don't we want to keep `--daemon` when async?

Reply:
This list is identical to the `args` variable defined above, so I just reused `args`.

Reply:
`args = [exe, "--daemon", self.path]` on line 463.

Reply:
Oh Isee, my bad. Should we remove `--daemon` for sync? Not sure if it really makes sense to launch it as a daemon process when using sync mode.

Reply:
Actually, do we even need to start a subprocess when we synchronously cache? Also, what will happen to pending variants (see line 425)? If I understand the code, an in-flight variant would be skipped, which would result in a potentially broken environment...

Reply:
We probably don't need it no.  I was more or less copying the logic from the original PR for that.

There also seems to be a fair amount of logic in the CLI entry point for gathering things for the cache, so I'm not sure if we can really run it without the CLI.  for instance, look at `rez/cli/pkg-cache` line 154.  That seems to be the primary business logic of caching without the daemon.  If we were to remove the subprocess, we should move that whole block into the PackageCache class, which feels maybe beyond the scope of what we were trying to do, but maybe not.  Let me know if you'd like to forge ahead with that strategy.